Replicator/EchoCanceler.hh

```diff
@@ -0,0 +1,90 @@
+//
+// EchoCanceler.hh
+//
+// Copyright 2025-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+//
+
+#pragma once
+#include "Base.hh"
+#include "Logging.hh"
+#include "ReplicatorTypes.hh"
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+C4_ASSUME_NONNULL_BEGIN
+
+namespace litecore::repl {
+
+    /** A set of {docID, revID} pairs that's used to avoid echoing a peer's own revisions back to it.
+        (There's actually one of these sets per collection, but only for collections that have
+        bidirectional continuous replication.)
```

Comment:
Regarding what Jianmin pointed out above, maybe just this comment is slightly inaccurate?